scripts/prepareDocs.js

```diff
@@ -1,15 +1,66 @@
 #!/usr/bin/env node
 const fs = require('fs');
 
-const analysis = require('../analysis.json');
-const elements = analysis.elements.filter((el) => el.privacy === 'public');
+function generateChangeEventName(property) {
+  return `${property.name}-changed`;
+}
 
-const result = {
-  // eslint-disable-next-line camelcase
-  schema_version: '1.0.0',
-};
+function findEvent(element, eventName) {
+  return element.events.find((event) => event.name === eventName);
+}
 
-result.mixins = analysis.mixins;
-result.elements = elements;
+/**
+ * Adds missing change event declarations to elements. The Polymer Analyzer
+ * only automatically generates these if the property is declared in the
+ * component class, but not if it is declared in a mixin. This fix scans
+ * for notifiable properties and adds respective change event declarations
+ * if they don't exist yet.
+ * @param analysis
+ */
+function addMissingChangeEventDeclarations(analysis) {
+  const elements = analysis.elements;
 
-fs.writeFileSync('./analysis.json', JSON.stringify(result, null, 2));
+  elements.forEach((element) => {
+    element.properties.forEach((property) => {
+      // Check if property is marked as notify, skip if not
+      const hasNotify = property && property.metadata && property.metadata.polymer && property.metadata.polymer.notify;
+      if (!hasNotify) {
+        return;
+      }
+      // Check if there is an existing event declarations, skip if exists
+      const changeEventName = generateChangeEventName(property);
+      const existingEvent = findEvent(element, changeEventName);
+      if (existingEvent) {
+        return;
+      }
+      // Add missing event
+      const eventDeclaration = {
+        name: changeEventName,
+        type: 'CustomEvent',
+        description: `Fired when the \`${property.name}\` property changes.`,
+        metadata: {},
+      };
+      element.events = element.events || {};
```

Comment:
Should this be an empty array instead, as you are using `push` below?

Reply:
Of course, fixed